app/src/main/java/com/mapbox/maps/testapp/examples/java/RuntimeStylingJavaActivity.java

```diff
@@ -0,0 +1,423 @@
+        addFillExtrusionLayer(style);
+        addFillExtrusionLight(style);
+
+        addImageSource(style);
+        addRasterLayer(style);
+
+        addLayerWithoutStyleExtension(style);
+
+        final VectorSource source = (VectorSource) SourceUtils.getSource(style, "composite");
+        Logger.e(TAG, "getSource: " + source);
+    }
+
+    private void addImage(Style style) {
+        final Drawable drawable = ContextCompat.getDrawable(this, R.drawable.android_symbol);
+        final Bitmap bitmap = DrawableKt.toBitmap(drawable, 64, 64, null);
+        final Expected<String, None> expected = style.addImage(IMAGE_ID, bitmap, true);
+        if (expected.isError()) {
+            Logger.e(TAG, expected.getError());
+        }
+        if (expected.isValue()) {
+            Logger.d(TAG, expected.getValue().toString());
+        }
+    }
+
+    private void addSymbolSource(Style style) {
+        final FeatureCollection featureCollection = FeatureCollection.fromJson(SYMBOL_SOURCE_FEATURECOLLECTION);
+        final GeoJsonSource geoJsonSource = new GeoJsonSource.Builder(GEOJSON_SOURCE_ID, source -> {
+        }).featureCollection(featureCollection)
+                .cluster(true)
+                .prefetchZoomDelta(1)
+                .build();
+        Logger.i(TAG, geoJsonSource.toString());
+        SourceUtils.addSource(style, geoJsonSource);
+        Logger.i(TAG, "prefetchZoomDelta :" + geoJsonSource.getPrefetchZoomDelta());
+    }
+
+    private void addSymbolLayer(Style style) {
+        final Expression textField = new Expression.FormatBuilder()
+                .formatSection("London", formatSectionBuilder -> {
+                    formatSectionBuilder.fontScale(1.0);
+                    formatSectionBuilder.textFont(TEXT_FONT);
+                    formatSectionBuilder.textColor(Color.RED);
+                    return null;
+                }).formatSection(image(literal("london-underground")), formatSectionBuilder -> {
+                    formatSectionBuilder.fontScale(0.9);
+                    return null;
+                }).formatSection("underground", formatSectionBuilder -> {
+                    formatSectionBuilder.fontScale(0.8);
+                    formatSectionBuilder.textFont(TEXT_FONT);
+                    formatSectionBuilder.textColor(Color.WHITE);
+                    return null;
+                }).build();
+
+        SymbolLayer symbolLayer = new SymbolLayer(SYMBOL_LAYER_ID, GEOJSON_SOURCE_ID);
+        symbolLayer.filter(eq(get("count"), literal(0)));
+        symbolLayer.iconImage(literal(IMAGE_ID));
+        symbolLayer.iconOpacity(subtract(literal(1.0), literal(0.6)));
+        symbolLayer.textField(textField);
+        symbolLayer.iconColor(Color.GREEN);
+        symbolLayer.textAnchor(TextAnchor.CENTER);
+        symbolLayer.iconAnchor(IconAnchor.BOTTOM);
+        symbolLayer.textIgnorePlacement(false);
+        symbolLayer.iconIgnorePlacement(false);
+        LayerUtils.addLayer(style, symbolLayer);
```

Comment:
shall we use StyleUtils in this case? I think we agreed to separate the extension functions to different files, shall we already start to use that practice gradually when improving Java compatibility?

Reply:
Do you mean we use same name `StyleUtils` for layer/ source/light/terrain?